app/views/distributions/_form.html.erb

```diff
@@ -42,7 +49,56 @@
     </fieldset>
   </div>
   <div class="card-footer">
-    <%= submit_button %>
-  </div>
+    <%= submit_button({}, { action: "click->distribution-confirmation#openModal" }) %>
   </div>
 <% end %>
+
+<%# Confirmation modal: See distribution_confirmation_controller.js for how this gets displayed and populated. %>
+<div id="distributionConfirmationModal"
+  class="modal confirm"
+  aria-labelledby="distributionConfirmationModal"
+  aria-hidden="true"
+  tabindex="-1"
+  data-bs-backdrop="static"
+  data-distribution-confirmation-target="modal">
+  <div class="modal-dialog modal-lg">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title">Distribution Confirmation</h5>
+        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+      </div>
+      <div class="modal-body">
+        <p class="lead">You are about to create a distribution for
+          <span class="fw-bolder fst-italic"
+            data-distribution-confirmation-target="partnerName"
+            data-testid="distribution-confirmation-partner"></span>
+          from
+          <span class="fw-bolder fst-italic"
+            data-distribution-confirmation-target="storageName"
+            data-testid="distribution-confirmation-storage"></span>
+          :
+        </p>
+
+          <table class="table">
+            <thead>
+              <tr>
+                <th>Item Name</th>
+                <th>Quantity</th>
+              </tr>
+            </thead>
+            <tbody data-distribution-confirmation-target="tbody">
+
+            </tbody>
+          </table>
+
+          <div class="message fs-5">
+            <p>Please confirm that the above list is what you meant to distribute.</p>
```

Comment:
@danielabar   Hrmm.   Now that I see it in action, I want a small adjustment here -- let's change "meant" to "want".  

 Reason:  We are displaying this before any error checking.   The past tense of "meant" implies you are pretty much finished.  "Want" is better for setting expectations of being done. (which they aren't,  if error checking hasn't happened yet)

Reply:
Updated the wording.